proposals/0000-no-toplevel-field-selectors.rst

```diff
@@ -38,28 +38,29 @@ bindings for other values.
 - namespaced lenses
 - overloaded-labels based accessors / lenses
 - row types
-- https://github.com/ghc-proposals/ghc-proposals/pull/158
+- `<https://github.com/ghc-proposals/ghc-proposals/pull/158>`_
 - ...
 
 Proposed Change Specification
 -----------------------------
 
 Record definition no longer defines toplevel selector functions when the
-``NoToplevelFieldSelectors`` extension is enabled.
+``NoFieldSelectors`` extension is enabled. The default is ``FieldSelectors``,
+which implies the current behavior, that selector functions are being generated.
 
 Record construction/update syntax and pattern matching will work as before, the
 disambiguation handled by ``DuplicateRecordFields``. The necessary selectors
-will be part of a ``Record(..)`` export, or can also be named one by one. They
+will be part of a ``Record(..)`` export, or can also be named individually. They
 always have to be associated with the data type though, because there is no more
-toplevel selector.
+toplevel selector (see `Example`_).
 
 A function for Template Haskell to go from a record field name to a selector for
 that field will be provided, because it's not possible anymore to go from field
 name directly to selector.
 
-A new `TH` function is added which takes two `Name` and returns an `Exp` which
-is equivalent to `a -> b`, where `a` is the record type specified by the second
-argument, and `b` the contents of the field specified by the first argument.
+A new ``TH`` function is added which takes a ``Name`` (for the constructor) and
+a ``String`` (for the field) and returns a ``Maybe Name``, just like
+``lookupValueName``.
```

Comment:
Oh, I see this is already in S2.2. Then I think a forward reference at the end of the last paragraph could replace this one entirely.

Reply:
Taking another look at this, it's not clear to me why you changed the new TH function to return a `Name` instead of an `Exp`. How would this work? The field selector does not necessarily exist anymore so you cannot refer to it by name. You could say the semantics are that it generates a local or top-level definition and returns that `Name`, but that seems a lot more cumbersome than just returning a selector as an `Exp` directly.

Reply:
Yeah, `Exp` makes more sense here.